src/constants.ts

```diff
@@ -2,27 +2,125 @@ import { Endpoints } from './models/core/Endpoints';
 
 export const defaultApiVersion = 20220511;
 
-export const defaultEndpoints: Required<Endpoints> = {
-  universalSearch: 'https://liveapi.yext.com/v2/accounts/me/search/query',
-  verticalSearch: 'https://liveapi.yext.com/v2/accounts/me/search/vertical/query',
-  questionSubmission: 'https://liveapi.yext.com/v2/accounts/me/createQuestion',
-  status: 'https://answersstatus.pagescdn.com',
-  universalAutocomplete: 'https://liveapi-cached.yext.com/v2/accounts/me/search/autocomplete',
-  verticalAutocomplete: 'https://liveapi-cached.yext.com/v2/accounts/me/search/vertical/autocomplete',
-  filterSearch: 'https://liveapi-cached.yext.com/v2/accounts/me/search/filtersearch',
-} as const;
+export enum CloudRegion {
+  US = 'us',
+  EU = 'eu',
+}
+
+export enum Environment {
+  PROD = 'prod',
+  SANDBOX = 'sbx',
+}
 
 /**
- * The endpoints to use for sandbox experiences.
+ * Provides methods for getting various endpoints.
  *
  * @public
  */
-export const SandboxEndpoints: Required<Endpoints> = {
-  universalSearch: 'https://liveapi-sandbox.yext.com/v2/accounts/me/search/query',
-  verticalSearch: 'https://liveapi-sandbox.yext.com/v2/accounts/me/search/vertical/query',
-  questionSubmission: 'https://liveapi-sandbox.yext.com/v2/accounts/me/createQuestion',
-  status: 'https://answersstatus.pagescdn.com',
-  universalAutocomplete: 'https://liveapi-sandbox.yext.com/v2/accounts/me/search/autocomplete',
-  verticalAutocomplete: 'https://liveapi-sandbox.yext.com/v2/accounts/me/search/vertical/autocomplete',
-  filterSearch: 'https://liveapi-sandbox.yext.com/v2/accounts/me/search/filtersearch',
-} as const;
\ No newline at end of file
+export class EndpointsProvider {
+  private readonly environment: Environment;
+  private readonly cloudRegion: CloudRegion;
+  private readonly overrideEndpoints?: Endpoints;
+
+  constructor(environment?: Environment, cloudRegion?: CloudRegion, overrideEndpoints?: Endpoints) {
```

Comment:
I'm not sure that overrideEndpoints is necessary here. Someone could manually set additional override outside of this class. for example: 
```
const providedEndpoints = new EndpointsProvider(
      Environment.SANDBOX,
      CloudRegion.US
).getEndpoints();

const customEndpoints = {
   ...proviedEndpoints,
   verticalSearch: '<verticalSearchEndpoint>'
}
```